File: InVehicleDevice/src/main/java/com/kogasoftware/odt/invehicledevice/view/fragment/utils/ViewDisabler.java
package com.kogasoftware.odt.invehicledevice.view.fragment.utils;

import android.os.Handler;
import android.view.View;

/**
 * ボタンなどの多重操作の防止
 */
public class ViewDisabler {
    public static void disable(final View view) {
        disable(view, 400);
    }

    public static void disable(final View view, Integer timeoutMillis) {
        view.setEnabled(false);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                view.setEnabled(true);
            }
        }, timeoutMillis);
    }
}

File: InVehicleDevice/src/main/java/com/kogasoftware/odt/invehicledevice/view/fragment/utils/ViewDisabler.kt
package com.kogasoftware.odt.invehicledevice.view.fragment.utils

import android.os.Handler
import android.view.View
import kotlin.jvm.JvmOverloads

/**
 * ボタンなどの多重操作の防止
 */
object ViewDisabler {
    @JvmOverloads
    @JvmStatic
    fun disable(view: View, timeoutMillis: Int = 400) {
        view.isEnabled = false
        Handler().postDelayed({ view.isEnabled = true }, timeoutMillis.toLong())
    }
}
